### PR TITLE
[DOCS] Update node descriptions for default distribution

### PR DESCRIPTION
**Pull request by lcawl**
This PR updates https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-node.html such that it's clearer that the initial examples don't apply to the default distribution. 

It requires https://github.com/elastic/docs/pull/917

It also cleans up some outstanding disparities in the dedicated node definitions related to https://github.com/elastic/elasticsearch/issues/25210

### CODE REVIEW

File: docs/reference/modules/node.asciidoc [hunk 1]
```diff
@@ -131,8 +132,9 @@ cluster.remote.connect: false <4>
 <4> Disable {ccs} (enabled by default).
 
 ifdef::include-xpack[]
-NOTE: These settings apply only when {xpack} is not installed. To create a
-dedicated data node when {xpack} is installed, see <<modules-node-xpack,{xpack} node settings>>.
+NOTE: By default, when you install {es}, {xpack} is installed.These settings
```
Comment (pmusa) [inline docs/reference/modules/node.asciidoc]: it is missing a space after `installed.`

File: docs/reference/modules/node.asciidoc [hunk 2]
```diff
@@ -159,8 +161,9 @@ cluster.remote.connect: false <4>
 <4> Disable {ccs} (enabled by default).
 
 ifdef::include-xpack[]
-NOTE: These settings apply only when {xpack} is not installed. To create a
-dedicated ingest node when {xpack} is installed, see <<modules-node-xpack,{xpack} node settings>>.
+NOTE: By default, when you install {es}, {xpack} is installed.These settings
```
Comment (pmusa) [inline docs/reference/modules/node.asciidoc]: same here.

**Review by pmusa — COMMENTED** [review]
I am not sure I am the right person to review this PR. From my understanding we were avoiding using the term `x-pack`, specially in the context of installing. The sentences are weird because x-pack is a gray area. You can use ML in basic (There is the data visualizer and import data though). Maybe calling out that the `node.ml` setting should not be used in the OSS version.

**Comment by lcawl** [comment]
These changes can be previewed here: https://doc-review-63bf6.firebaseapp.com/es1/modules-node.html

**Pull request merged**